design-documents/catalog-domain/catalog-service.md

```diff
@@ -0,0 +1,204 @@
+# Catalog Service
+
+## Purpose of Catalog Service
+Catalog has a single purpose of returning product information that was intended to be shown at the storefront, both product listing or PDP.
+Basically saying Catalog will answer on questions:
+* How the product should look like to satisfy storefront requirements.
+* How client of the Catalog can request products.
+
+## Usage examples
+
+### Direct call
+Catalog service can be involved directly through the gateway.
+![Dirrect call](images/catalog-usage-01-01.png)
+
+A good example of such services is sales channels integrations that have to synchronize across connected channels.
+
+### Routing
+Clients who want to retrieve product data by URL.
+![Routing](images/catalog-usage-01-02.png)
+
+Literally any application.
+
+
+### Listings and search
+
+Category page, search results, layered navigation.
+![Routing](images/catalog-usage-01-03.png)
+
+## Defining the request
+
+See [Product Data Object](product-data-object.md)
+
+* Different scenarios require different data of the same product.
+* Catalog service is responsible for returning the most accurate product representation. 
+* Catalog service does not perform searching but has to perform a basic data filtering.
+* The purpose of filtering is reducing data before sending it to a client.
+Example of such data reduction are 
+    * On Listing Page, we need only attributes applicable for the listing.
+    * On Product Details Page we need to return variations only for the selected options.
+
+
+
+### JSON
+
+```json
+{
+  "query": "products",
+  "filters": [
+    {
+      "field":  "id",
+      "values": [
+        "1652b2fe-f758-11e9-8f0b-362b9e155667",
+        "1652b588-f758-11e9-8f0b-362b9e155667",
+        "1652b7c2-f758-11e9-8f0b-362b9e155667",
+        "1652b8f8-f758-11e9-8f0b-362b9e155667",
+        "1652bb00-f758-11e9-8f0b-362b9e155667",
+        "1652bc40-f758-11e9-8f0b-362b9e155667",
+        "1652be5c-f758-11e9-8f0b-362b9e155667",
+        "1652bf88-f758-11e9-8f0b-362b9e155667",
+        "1652c186-f758-11e9-8f0b-362b9e155667"
+      ]
+    }
+  ],
+  "fields": [
+    {"field": "name"},
+    {"field": "sku"},
+    {
+      "document": "priceRange",
+      "fields": [
+        {
+          "document": "minimumPrice",
+          "fields": [
+            {"field": "finalPrice"}
+          ]
+        }
+      ]
+    },
+    {
+      "document": "attributes",
+      "filters": [
```

Comment:
What the reason to have a "filters" node here? In GQL request we ask to provide different attributes (fields). And the proposed approach is to complex and do not provide any visible benefits.
Why we can't just ask a list of fields that we want to receive from service using don-notation for nested attributes? E.g.
```
"fields" : [
    "name",
    "sku",
    "priceRange.minimumPrice.finalPrice"
] 
```